modules/experimental-layers/src/gpu-grid-layer/gpu-grid-layer.js

```diff
@@ -61,33 +61,48 @@ export default class GPUGridLayer extends CompositeLayer {
     };
   }
 
-  updateState({oldProps, props, changeFlags}) {
-    const reprojectNeeded = this.needsReProjectPoints(oldProps, props);
-
-    if (changeFlags.dataChanged || reprojectNeeded) {
-      // project data into hexagons, and get sortedBins
-      this.getLayerData();
+  updateState(opts) {
+    const aggregationFlags = this.getAggregationFlags(opts);
+    if (aggregationFlags) {
+      // project data into grid cells
+      this.getLayerData(aggregationFlags);
     }
   }
 
-  needsReProjectPoints(oldProps, props) {
-    return (
-      oldProps.cellSize !== props.cellSize ||
-      oldProps.gpuAggregation !== props.gpuAggregation ||
-      oldProps.getPosition !== props.getPosition
-    );
+  getAggregationFlags({oldProps, props, changeFlags}) {
+    let aggregationFlags = null;
+    if (
+      changeFlags.dataChanged ||
+      oldProps.getPosition !== props.getPosition ||
```

Comment:
This is an anti-pattern for accessors. Check `changeFlags.updateTriggers.getPosition` instead.

Reply:
Updated, opened an issue to track all other layers :  #2399